File: two pointers/Valid Palindrome.py
def isPalindrome(s: str) -> bool:
    l, r = 0, len(s) - 1

    while l <= r:
        left, right = s[l].lower(), s[r].lower()
        print(left, right)
        if not left.isalnum():
            l += 1
        elif not right.isalnum():
            r -= 1
        elif left != right:
            return False
        else:
            l += 1
            r -= 1

    return True

print(isPalindrome("A man, a plan, a canal: Panama"))

File: two pointers/Valid Palindrome.cs
public class SolutionIsPalindrome
{
    public bool IsPalindrome(string s)
    {
        int l = 0;
        int r = s.Length - 1;

        while (l < r)
        {
            while (l < r && !char.IsLetterOrDigit(s[l]))
                l++;
            while (l < r && !char.IsLetterOrDigit(s[r]))
                r--;

            char left = char.ToLower(s[l]);
            char right = char.ToLower(s[r]);

            if (left != right)
                return false;
            l++;
            r--;
        }

        return true;
    }
}
